crates/sparrow-runtime/src/prepare/preparer.rs

```diff
@@ -24,55 +34,148 @@ pub enum Error {
     SortingBatch,
     #[display(fmt = "unrecognized time unit")]
     UnrecognizedTimeUnit(String),
+    #[display(fmt = "invalid url '{_0:?}'")]
+    InvalidUrl(String),
+    #[display(fmt = "internal error")]
+    Internal,
 }
 
 impl error_stack::Context for Error {}
 
+/// The current prepare version
+const PREPARE_VERSION: i64 = 0;
+
 pub struct Preparer {
     prepared_schema: SchemaRef,
-    time_column_name: String,
-    subsort_column_name: Option<String>,
+    table_config: Arc<TableConfig>,
     next_subsort: AtomicU64,
-    key_column_name: String,
     time_multiplier: Option<i64>,
+    object_stores: Arc<ObjectStoreRegistry>,
 }
 
 impl Preparer {
     /// Create a new prepare produce data with the given schema.
     pub fn new(
-        time_column_name: String,
-        subsort_column_name: Option<String>,
-        key_column_name: String,
+        table_config: Arc<TableConfig>,
         prepared_schema: SchemaRef,
         prepare_hash: u64,
         time_unit: Option<&str>,
+        object_stores: Arc<ObjectStoreRegistry>,
     ) -> error_stack::Result<Self, Error> {
         let time_multiplier = time_multiplier(time_unit)?;
         Ok(Self {
             prepared_schema,
-            time_column_name,
-            subsort_column_name,
+            table_config,
             next_subsort: prepare_hash.into(),
-            key_column_name,
             time_multiplier,
+            object_stores,
         })
     }
 
     pub fn schema(&self) -> SchemaRef {
         self.prepared_schema.clone()
     }
 
+    /// Prepare a parquet file.
+    ///
+    /// - This computes and adds the key columns.
+    /// - This sorts the files by time, subsort and key hash.
+    /// - This adds or casts columns as needed.
+    /// - This produces multiple parts if the input file is large.
+    /// - This produces metadata files alongside data files.
+    pub async fn prepare_parquet(
+        &self,
+        path: &std::path::Path,
+    ) -> error_stack::Result<Vec<PreparedFile>, Error> {
+        // TODO: Support Slicing
+
+        // Prepared files are stored in the following format:
+        // file:///<cwd>/tables/<table_uuid>/prepared/<prepare_version>/<uuid>/part-<n>.parquet
```

Comment:
I don't think we *need* them to be stored in that format anymore. Without Wren we don't have tables anymore, etc.

Reply:
We still have a table uuid id though, right, and can have multiple tables per session? What parts of this could we drop?

Reply:
I mean, in theory we could (for now) put them all in a just a "<prepare_prefix>/<uuid>.parquet" path. No reason to associate them with specific prepare versions, etc.

At the least, we could maybe do `<cwd>/tables/<table_uuid>/prepared/<uuid>.parquet` may be enough? 

I think the first UUID was so that we could have data tokens exclude the file, which we're currently not anticipating supporting. Additionally, changing prepare version may not matter since this is ephemeral.

Reply:
Removed the `prepare_version`. 

I think there is benefit in keeping the `table_uuid` and `<uuid>/part_n.parquet` format. For debugging purposes, the user is able to (more easily, though not entirely) inspect their prepared files. Probably nice if we would include the actual table name in the path, but for now I'm fine with it as is until we decide during state management design.